ClientApp/viking-vault/src/Router.tsx

```diff
@@ -44,8 +45,31 @@ class Router extends React.Component<any, IState> {
     private makeRedirect() {
         if(sessionStorage.getItem("Authentication-Token") === null)
             return <Redirect to="/login" />;
-        else
-            return <Redirect to={this.state.isAdmin? "/admin" : "/user"} />;
+        else {
+            if(this.state.isAdmin === null) {
+                setTimeout(() =>{
+                    return <Redirect to={this.state.isAdmin? "/admin" : "/user"} />;
+                }, 100);
+            }
+            else return <Redirect to={this.state.isAdmin? "/admin" : "/user"} />;
+        }
+    }
+
+    private makeAdminRedirect()
+    {
+        if(this.state.isAdmin === null)
+        {
+            setTimeout(() => {
+                if(this.state.isAdmin)
+                    return <AdminPage/>;
+                else return <Redirect to = "/login" />
+            },200);
```

Comment:
Same here, isn't the 200ms too small to be perceivable, couldn't you simply make the redirect instantly?

Reply:
Same explanation as before.